assets/js/base/components/cart-checkout/address-form/address-form.tsx

```diff
@@ -229,9 +262,19 @@ const AddressForm = ( {
 						onChange={ ( newValue: string ) =>
 							onChange( {
 								...values,
-								[ field.key ]: newValue,
+								[ field.key ]:
+									field.key === 'postcode'
+										? newValue.trimStart().toUpperCase()
+										: newValue,
```

Comment:
When the input field is `postcode`, then remove leading spaces and convert lowercase to uppercase characters on input. This part ensures that the client-side postcode validation of WooCommerce Blocks, is in sync with [the server-side postcode validation of WooCommerce](https://woocommerce.github.io/code-reference/files/woocommerce-includes-wc-formatting-functions.html#source-view.1001-1012).

Reply:
@mikejolley Based on [our internal discussion](p1678276921630599/1678276379.331539-slack-C8X6Q7XQU), I've updated the PR. Would you mind reviewing it again?